src/components/modal/ModalDialog.js

```diff
@@ -110,13 +144,19 @@ class ModalDialog extends React.Component<Props> {
             divProps['aria-describedby'] = `${this.modalID}-desc`;
         }
 
+        let modalHeaderClasses = 'modal-header';
+        if (showBackButton) {
+            modalHeaderClasses += ' modal-header--with-back';
+        }
+
         return (
             <div ref={modalRef} className={classNames('modal-dialog', className)} {...divProps}>
-                <div className="modal-header">
+                <div className={modalHeaderClasses}>
                     <h2 className="modal-title" id={`${this.modalID}-label`}>
                         {title}
                     </h2>
                 </div>
+                {showBackButton && this.renderBackButton()}
```

Comment:
Should this be inside the header, since that's its visual/structural parent? It would allow us to use the adjacent sibling selector to avoid the `--with-back` modifier:

```
{this.renderBackButton()}
<h2 className="modal-title" id={`${this.modalID}-label`}>
    {title}
</h2>
```

```
.modal-back-btn + .modal-title {
    margin-left: 20px;
}